lib/core/src/display/navigation/navigator.rs

```diff
@@ -0,0 +1,94 @@
+use super::navigator_events::NavigatorEvents;
```

Comment:
This implementation lacks one input method - touchpad. It works with a mouse only now. We need touchpad support as well. I believe that would not be hard to add here, as the events are handled the same way here. We want pinch event to be mapped to zoom, while dragging with two fingers mapped to pan.